rust/src/gildedrose.rs

```diff
@@ -30,56 +30,54 @@ impl GildedRose {
         GildedRose { items }
     }
 
+    pub fn increase_by_one(item: &mut Item) {
+        if item.quality < 50 {
+            item.quality = item.quality + 1
+        }
+    }
+
     pub fn update_quality(&mut self) {
         for item in &mut self.items {
-            if item.name != "Aged Brie" && item.name != "Backstage passes to a TAFKAL80ETC concert"
-            {
-                if item.quality > 0 {
-                    if item.name != "Sulfuras, Hand of Ragnaros" {
-                        item.quality = item.quality - 1;
+            match item.name.as_str() {
```

Comment:
Nice use of match - I couldnt work out how to use item.name in match as I got some type errors.

Reply:
Also some nice separation of the logic in here to keep things cleaner. Will also aid updating and adding logic if new requirements come along.